main/src/cgeo/geocaching/calculator/CoordinatesCalculateUtils.java

```diff
@@ -0,0 +1,163 @@
+
+        String coordinateChars = ""; // All the characters that appear in the coordinate representation.
+
+        char latHem = 'N';
+        if (latText.length() > 0) {
+            final char first = latText.charAt(0);
+            if (first == 'N' || first == 'S') {
+                latHem = first;
+                coordinateChars = coordinateChars.concat(latText.substring(1));
+            } else {
+                coordinateChars = coordinateChars.concat(latText);
+            }
+        }
+
+        char lonHem = 'W';
+        if (lonText.length() > 0) {
+            final char first = lonText.charAt(0);
+            if (first == 'E' || first == 'W' || first == 'O') {
+                lonHem = first;
+                coordinateChars = coordinateChars.concat(lonText.substring(1));
+            } else {
+                coordinateChars = coordinateChars.concat(lonText);
+            }
+        }
+
+        List<VariableData> equData = new ArrayList<>();
+        equData = CoordinatesCalculateUtils.sortVariables(equData, variableDataList, coordinateChars, new CaseCheck(true));
+
+        String equationStrings = "";
+        for (final VariableData equ : equData) {
+            equationStrings = equationStrings.concat(equ.getExpression());
+        }
+
+        // replace the old free variables list with a newly created ones.
+        List<VariableData> freeVarData = new ArrayList<>();
+        freeVarData = CoordinatesCalculateUtils.sortVariables(freeVarData, variableDataList, equationStrings, new CaseCheck(false));
+
+        final List<ButtonData> butData = new ArrayList<>();
+
+        final CalcState calcState = new CalcState(Settings.CoordInputFormatEnum.Plain,
+            latText,
+            lonText,
+            latHem,
+            lonHem,
+            butData,
+            equData,
+            freeVarData,
+            variableDataList);
+
+        return calcState;
+    }
+
+    /**
+     * Find if a variable exists in the supplied list with the given name
+     *
+     * @param name name to search for
+     * @param list list of variables
+     * @return first occurrence of the variable if it can found, 'null' otherwise
+     */
+    private static VariableData getVariable(final char name, final List<VariableData> list, final boolean remove) {
+        for (final VariableData equ : list) {
+            if (equ.getName() == name) {
+                if (remove) {
+                    list.remove(equ);
+                }
+                return equ;
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Find if variable data exists in the supplied list with the given name
+     *
+     * @param name name to search for
+     * @return first occurrence of the data if it can found, 'null' otherwise
+     */
+    private static VariableData findAndRemoveData(final char name, final List<VariableData> list) {
```

Comment:
Isn't this the same as calling "getVariable" with "remove=true"?

Reply:
Seems so.
I see 3 possibilities:
1. call getVariable(...false) in the method findAndRemoveData
2. replace the whole method findAndRemoveData with getVariable(...false)
3. leave both methods and remove the parameter remove from getVariable and adapt all calls of getVariable(....true)  with findAndRemoveData
4. combination out of all having 3 functions: 
  - findVariableData without boolean
  - findVariableDataAndRemove
  - findVariableData with boolean: the first two calls the third one

I am not sure, which one I prefer, but variant 1 is the less preferred

BTW isn't there any package in java which can handle that?

Reply:
With your next comment, probably variant 3 or 4 will be best

Reply:
I like 3 best :-)

Reply:
Done